Source/Particles/Deposition/ChargeDeposition.H

```diff
@@ -213,4 +214,124 @@ void doChargeDepositionShapeN (const GetParticlePosition& GetPosition,
 #endif
 }
 
+template <typename PC>
+void DepositChargeDoIt (typename PC::ParIterType& pti,
```

Comment:
Somewhere in this API, `WarpX::do_device_synchronize` is called (probably in a lower-level call).

We need to factor this out as an option, too.